src/AutoRest.CSharp/Common/Output/Models/Types/EnumType.cs

```diff
@@ -58,15 +58,19 @@ public EnumType(InputEnumType input, string defaultNamespace, string defaultAcce
             ValueType = typeFactory.CreateType(input.EnumValueType);
             IsStringValueType = ValueType.Equals(typeof(string));
             IsIntValueType = ValueType.Equals(typeof(int)) || ValueType.Equals(typeof(long));
+            IsFloatValueType = ValueType.Equals(typeof(float));
+            IsNumericValueType = IsIntValueType || IsFloatValueType;
             SerializationMethodName = IsStringValueType && IsExtensible ? "ToString" : $"ToSerial{ValueType.Name.FirstCharToUpperCase()}";
```

Comment:
I assume this is a potential refactor here to let it have a `IEnumerable<Method>` here instead of just having the serialization method name.

Reply:
Yes. I have it in consolidation PR where it is done for both `ModelTypeProvider` and `MgmtObjectType`.